getnacelle/nacelle-js
# feat(ENG-6319): sanity plugin nacelle references

## WHY are these changes introduced?

[ENG-6319](https://nacelle.atlassian.net/browse/ENG-6319) 



## WHAT is this pull request doing?

Adds support for Nacelle Reference objects per [ADR040](https://nacelle.atlassian.net/wiki/spaces/DD/pages/1749712897/ADR+040+-+Structured+Commerce+Data+References+in+Content)

- Adds a second input/part to plugin export `sanity.json`
- Refactors/renames/moves reusable components to `shared` folder
- Renames top level component `nacelleData` to match package exports
- updates HF queries to include `__typename`
- Creates new top level component/schema type `nacelleReference` for storing reference objects


## How to Test
For local testing I used [Sanity Reference Store](https://github.com/getnacelle/nacelle-reference-store-sanity-studio) repo.

- Pull down this branch, build and link the sanity plugin to local Sanity studio project
- Update a content schema in Sanity studio project to use `nacelleReference` type (see below).  For this example we used `schemas/product/_content.js`
<img width="296" alt="Screen Shot 2022-08-22 at 10 25 04 AM" src="https://user-images.githubusercontent.com/24395931/185971189-3d13ee57-c3be-4360-ae0e-ecc8ea2f5d3d.png">
- Update the piece of content so that it links to a new product
<img width="1111" alt="Screen Shot 2022-08-22 at 10 27 04 AM" src="https://user-images.githubusercontent.com/24395931/185971443-c1e1083b-c02a-4d81-9b34-c00f899812e1.png">
- Publish content, open the vision tab, and query for it.  You should see the `NacelleReference` object stored in the response.
<img width="1071" alt="Screen Shot 2022-08-22 at 10 28 01 AM" src="https://user-images.githubusercontent.com/24395931/185971715-ac7ac945-d944-44d5-bcbd-133c5c5265f3.png">

Notes:  The Sanity UI for both of the inputs intentionally looks identical.  We could show _what_ is being stored, but it felt a bit unnecessary.  Right now the saving of the object occurs behind the scenes.
